typed/typed.go

```diff
@@ -187,7 +201,41 @@ func (tv TypedValue) RemoveItems(items *fieldpath.Set) *TypedValue {
 }
 
 // ExtractItems returns a value with only the provided list or map items extracted from the value.
-func (tv TypedValue) ExtractItems(items *fieldpath.Set) *TypedValue {
+func (tv TypedValue) ExtractItems(items *fieldpath.Set, opts ...extractItemsOption) *TypedValue {
+	options := &extractItemsOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	if options.appendKeyFields {
+		tvPathSet, err := tv.ToFieldSet()
+		if err != nil {
```

Comment:
What if we do `if err == nil { <Add the logic to set the keyname here> }` ?
This way I think we could get rid of the `goto`. 
Wdyt?

Reply:
Sounds good. Updated.